### PR TITLE
[MM-12588] Complete migration of channel_actions to redux actions 

### PR DESCRIPTION
**Pull request by AGMETEOR**
**Summary**

MM-12588 - Complete migration of channel_actions to redux actions 

**Ticket Link**
[https://mattermost.atlassian.net/browse/MM-12588](https://mattermost.atlassian.net/browse/MM-12588)

Fixes https://github.com/mattermost/mattermost-server/issues/11442

### CODE REVIEW

**Comment by hanzei** [comment]
Hey @AGMETEOR,

I've restarted the build and it now works fine. It might help if you login into circleci. See https://community.mattermost.com/core/pl/xe6nb7u81brtbdda4koo4s1zrc for more information.

File: components/suggestion/search_channel_with_permissions_provider.jsx [hunk 1]
```diff
@@ -149,7 +153,7 @@ export default class SearchChannelWithPermissionsProvider extends Provider {
             return;
         }
 
-        const channelsAsync = ChannelActions.autocompleteChannelsForSearch(teamId, channelPrefix)(store.dispatch, store.getState);
+        const channelsAsync = this.autocompleteChannelsForSearch(teamId, channelPrefix);
```
Comment (saturninoabril) [inline components/suggestion/search_channel_with_permissions_provider.jsx]: Can you check if this is correct?  The previous one is from the mattermost-redux action which has `(teamId, channelPrefix)` params but the new one is from `actions/channel_action` which should have `(term, success, error)`.

File: actions/channel_actions.jsx [hunk 2]
```diff
@@ -113,34 +109,39 @@ export function searchMoreChannels(term) {
...
+            return;
+        }
+
+        const {data, error: err} = await dispatch(ChannelActions.autocompleteChannels(teamId, term));
+        if (data && success) {
+            success(data);
+        } else if (err && error) {
+            error({id: err.server_error_id, ...err});
+        }
+    };
```
Comment (saturninoabril) [inline actions/channel_actions.jsx]: I think we should move here those that access `store` (getState and dispatch) from ``ChannelMentionProvider`` (on success).
Please also consider the one at ``ChannelProvider``.

**Comment by AGMETEOR** [comment]
@saturninoabril sorry for the delayed response. I am now going to work on your feedback

File: components/suggestion/search_channel_with_permissions_provider.jsx (hunk 1 above)
Comment (AGMETEOR) [reply components/suggestion/search_channel_with_permissions_provider.jsx]: Hello @saturninoabril , if you look at line `107`, the constructor is passed a `channelSearchFunc`, that function is passed from the component consuming this provider and ultimately, it is `autoCompleteChannelsSearch` from mattermost redux.

File: actions/channel_actions.jsx (hunk 2 above)
Comment (AGMETEOR) [reply actions/channel_actions.jsx]: Hi @saturninoabril , I am not fully understanding what you mean here. I am kindly requesting that you elaborate more on `those` that I should move.

File: components/suggestion/search_channel_with_permissions_provider.jsx (hunk 1 above)
Comment (saturninoabril) [reply components/suggestion/search_channel_with_permissions_provider.jsx]: Oh I see.  The confusing part on me was I assumed that the `channelSearchFunc` passed into `SearchChannelWithPermissionsProvider` and `SearchChannelProvider` are the same (which I only realized now that it's different).

File: actions/channel_actions.jsx (hunk 2 above)
Comment (saturninoabril) [reply actions/channel_actions.jsx]: I'm referring to the following:
- [getMyChannelMemberships(store.getState())](https://github.com/mattermost/mattermost-webapp/blob/95730d6517abf279f3d3bf38f4b62841bc1cf134/components/suggestion/channel_mention_provider.jsx#L161)
- [getChannel(store.getState(), item.id)](https://github.com/mattermost/mattermost-webapp/blob/95730d6517abf279f3d3bf38f4b62841bc1cf134/components/suggestion/channel_mention_provider.jsx#L177)
- [store.dispatch()](https://github.com/mattermost/mattermost-webapp/blob/95730d6517abf279f3d3bf38f4b62841bc1cf134/components/suggestion/channel_mention_provider.jsx#L205)

But I think that should be dealt with separately.

**Comment by hanzei** [comment]
@AGMETEOR Would you please merge `master` into your branch?

**Comment by AGMETEOR** [comment]
@hanzei I have updated my branch with master. Just some blocked steps in the CI pipeline now.

**Comment by hanzei** [comment]
Thanks @AGMETEOR :+1:

**Pull request merged**